typing/typedecl.ml

```diff
@@ -341,7 +339,6 @@ let shape_map_cstrs =
 
 let transl_declaration env sdecl (id, uid) =
   (* Bind type parameters *)
-  Ctype.with_local_level begin fun () ->
```

Comment:
Just business as usual. The first part of the original `~post` was doing generalizations. They are now automatic. The second part is to be done after generalization, so it cannot be inside `~before_generalize`.
Since the original code used `with_local_level_if close`, this post processing must only be done if `closed = true`.
BTW, this is a direct translation; I have no idea why this code uses `TyVarEnv.instance_poly_univars` discarding its result. This doesn't really seem to make sense.